### PR TITLE
Enhance trading logic and improve UI components

### PR DESCRIPTION
**Pull request by jinming0618**
This pull request refactors how maximum trade size checks are handled across the perps trading UI, moving from shared, direction-agnostic validation logic to per-direction checks that are applied at the point of order submission. The main change is the adoption of a new `useDirectionMaxGate` hook, which enforces buy/sell-specific maximum trade size limits directly within each trading container's order placement handler. This results in cleaner validation logic and more accurate enforcement of trading constraints for each order direction.

**Trading logic improvements:**

- Replaced shared max trade size checks in form validation with per-direction checks using the new `useDirectionMaxGate` hook in all trading containers, including Limit, Market, TWAP, Take/Stop Limit, and Take/Stop Market. The hook is now imported and called in each relevant file, and the old validation logic that used a single effective max value has been removed. [[1]](diffhunk://#diff-5800c74efd60587593df3ec8cc90ff992dc062f94c9d232d665d27f2bdb8e32dR35) [[2]](diffhunk://#diff-d6f5bc716198dba279e6d2d9e8e52fe574d009fa54817075d3d847b1e70b8030R21) [[3]](diffhunk://#diff-78f583a9821895684044110596af23a700d557f9c8b7c00e9f996db2b6d0f634R17) [[4]](diffhunk://#diff-ecdec10db35ad667ba3914521f96119fb899ac6e7eeee16594c04d514e1c142cR21) [[5]](diffhunk://#diff-a0e180780b08933d5cd5103df32badd675548e6a0c9009b7f0f8dc16d5add6bcR20)

- Updated order submission handlers (e.g., `handleOrderClick`, `handleSubmitClick`, etc.) in each trading container to call `checkDirectionMax(isBuy)` before proceeding, ensuring that direction-specific trade size limits are enforced right before order confirmation. [[1]](diffhunk://#diff-5800c74efd60587593df3ec8cc90ff992dc062f94c9d232d665d27f2bdb8e32dR781) [[2]](diffhunk://#diff-d6f5bc716198dba279e6d2d9e8e52fe574d009fa54817075d3d847b1e70b8030R487-R494) [[3]](diffhunk://#diff-78f583a9821895684044110596af23a700d557f9c8b7c00e9f996db2b6d0f634R390) [[4]](diffhunk://#diff-ecdec10db35ad667ba3914521f96119fb899ac6e7eeee16594c04d514e1c142cR406)

**Validation logic cleanup:**

- Removed the old direction-agnostic max trade size checks from the shared form validation logic in each container, simplifying the validation code and delegating responsibility to the new per-direction gate. [[1]](diffhunk://#diff-5800c74efd60587593df3ec8cc90ff992dc062f94c9d232d665d27f2bdb8e32dL344-L359) [[2]](diffhunk://#diff-d6f5bc716198dba279e6d2d9e8e52fe574d009fa54817075d3d847b1e70b8030L113-L126) [[3]](diffhunk://#diff-78f583a9821895684044110596af23a700d557f9c8b7c00e9f996db2b6d0f634L200-L215) [[4]](diffhunk://#diff-ecdec10db35ad667ba3914521f96119fb899ac6e7eeee16594c04d514e1c142cL234-L252)

- Updated the dependency arrays for validation-related `useMemo` hooks to remove now-unnecessary dependencies on max trade size and reduceOnly flags. [[1]](diffhunk://#diff-5800c74efd60587593df3ec8cc90ff992dc062f94c9d232d665d27f2bdb8e32dL376-R382) [[2]](diffhunk://#diff-d6f5bc716198dba279e6d2d9e8e52fe574d009fa54817075d3d847b1e70b8030L144-L146) [[3]](diffhunk://#diff-78f583a9821895684044110596af23a700d557f9c8b7c00e9f996db2b6d0f634L233-R227) [[4]](diffhunk://#diff-ecdec10db35ad667ba3914521f96119fb899ac6e7eeee16594c04d514e1c142cL271-R263)

**Documentation and comments:**

- Updated comments throughout the affected files to clarify that the per-direction max gate now lives at the order button handlers, and that the form validation logic is direction-agnostic. [[1]](diffhunk://#diff-5800c74efd60587593df3ec8cc90ff992dc062f94c9d232d665d27f2bdb8e32dL324-R332) [[2]](diffhunk://#diff-d6f5bc716198dba279e6d2d9e8e52fe574d009fa54817075d3d847b1e70b8030L97-R98) [[3]](diffhunk://#diff-78f583a9821895684044110596af23a700d557f9c8b7c00e9f996db2b6d0f634L163-R164) [[4]](diffhunk://#diff-ecdec10db35ad667ba3914521f96119fb899ac6e7eeee16594c04d514e1c142cL209-R211) [[5]](diffhunk://#diff-a0e180780b08933d5cd5103df32badd675548e6a0c9009b7f0f8dc16d5add6bcL89-R90)

**Testing:**

- Added new tests to ensure that the `assumeSufficientMargin` logic for cross margin orders floors or preserves balances as expected, and that the new validation logic behaves correctly in edge cases.

This refactor improves maintainability and correctness by ensuring that buy and sell trade size limits are enforced precisely where they matter, and by reducing duplicated validation logic across the codebase.

### CODE REVIEW

**Comment by ddddhm1234** [comment]
# PR扫描结果
## 依赖变更

无依赖变更

**Comment by ddddhm1234** [comment]
# PR扫描结果
## 依赖变更

无依赖变更

**Comment by ddddhm1234** [comment]
# PR扫描结果
## 依赖变更

无依赖变更

File: src/ui/views/DesktopPerps/components/TradingPanel/utils.ts [hunk 1]
```diff
@@ -65,6 +66,24 @@ export const calcAmountFromPercentage = (
...
+ * size, so buy and sell gate on different limits — each button must check
+ * its own side. Slider input is exempt: its per-direction size is derived
+ * from that side's max and can never exceed it. A max of 0 means "unknown
+ * or not tradable" and is left to other gates (e.g. the reduce-only locks).
+ */
+export const exceedsDirectionMax = (
+  positionSize: PositionSize,
+  directionMax: string | number | undefined
+): boolean => {
+  if (positionSize.inputSource === 'slider') return false;
```
[the remark this answers is not in the document]
Comment (jinming0618) [reply src/ui/views/DesktopPerps/components/TradingPanel/utils.ts:82]: is not very important, hyperliquid will check in api submit

**Review by cs1707 — APPROVED** [review]
